### ISSUE TITLE
`LSTM` 학습시키기

### ISSUE THREAD

**Issue by eubinecto** [eubinecto/the-clean-rnns#16]
## To-do's
- [x] ainze에서 학습 진행, 로스가 RNN보다 더 줄어드는지 확인

**Comment by eubinecto** [comment]
## 로스가 크게 줄어들지는 않는다
https://wandb.ai/eubinecto/the-clean-rnns/runs/2auoujj3?workspace=user-eubinecto

애초에 LSTM도 0으로 converge하지를 않는다. 왜일까?  어찌되었든 train_loss는 0에 수렴해야하는데..
 이부분에 대한 해결이 필요하다.

**Comment by eubinecto** [comment]
## 질문 - 초기 가중치 설정을 잘못한 것인가?
### 실험
xavier  uniform으로 초기 가중치를 업데이트 해보자:
```python
    def on_train_start(self):
        # deep models should be initialised with so-called "Xavier initialisation"
        # refer to: https://proceedings.mlr.press/v9/glorot10a/glorot10a.pdf
        for param in tqdm(self.parameters(), desc="initialising weights..."):
            if param.dim() > 1:
                torch.nn.init.xavier_uniform_(param)


```

**Comment by eubinecto** [comment]
## 질문 - 데이터의 1%도 학습하지 못하는가?

### 실험
데이터의 1%에만 핏을 해보자
```shell
python3 run_train.py eubinecto lstm_for_classification --limit_train_batches=0.1 --limit_val_batches=0.1
```

**Comment by eubinecto** [comment]
## `lr`가 너무 크거나 너무 작은 것이 문제인가?

**Comment by eubinecto** [comment]
##  padding strategy가 문제인가?
### 실험?
post-padding -> pre-padding

**Comment by eubinecto** [comment]
## 다시 학습을 진행한 결과

모델 | f1 |  링크
rnn |   <img width="443" alt="image" src="https://user-images.githubusercontent.com/56193069/162093011-b239a9ca-473b-4cf9-8d89-e359c8a87071.png"> | https://wandb.ai/eubinecto/the-clean-rnns/runs/40ca3shv?workspace=user-eubinecto

lstm |  <img width="441" alt="image" src="https://user-images.githubusercontent.com/56193069/162093056-8034d94b-24ca-440b-abd4-a560805283d0.png"> | https://wandb.ai/eubinecto/the-clean-rnns/runs/25wm1ome?workspace=user-eubinecto
